libyul/AsmParser.cpp

```diff
@@ -249,6 +274,8 @@ ForLoop Parser::parseForLoop()
 	forLoop.pre = parseBlock();
 	forLoop.condition = make_unique<Expression>(parseExpression());
 	forLoop.post = parseBlock();
+	bool outsideForLoopBody = exchange(m_insideForLoopBody, true);
```

Comment:
Oh! I needed such a function earlier :)

Reply:
But actually, I think two separate statements would be easier to read in this case...

Reply:
I still don't think this is worth it.